yt_dlp/extractor/niconico.py

```diff
@@ -1004,6 +1001,30 @@ def _real_extract(self, url):
                     recv = recv[:100] + '...'
                 self.write_debug(f'Server said: {recv}')
 
+        ws.close()
+
+        formats = self._extract_m3u8_formats(m3u8_url, video_id, ext='mp4', live=is_live)
+        for fmt, q in zip(formats, reversed(qualities[1:])):
+            fmt.update({
+                'format_id': q,
+                'protocol': 'niconico_live',
+            })
+            yield fmt
```

Comment:
formats:

```
[
  {
    "format_id": "499",
    "tbr": 499.2,
  },
  {
    "format_id": "3900",
    "tbr": 3900.0,
  },
  {
    "format_id": "249",
    "tbr": 249.6,
  },
  {
    "format_id": "1300",
    "tbr": 1300.0,
  },
  {
    "format_id": "2600",
    "tbr": 2600.0,
  }
]

```

qualities:

```
['abr', 'super_high', 'high', 'normal', 'low', 'super_low', 'audio_high']

```

The "format_id" is wrong.

I think we can use some "changeStream" requests to get the m3u8 URL of each stream, including audio. The downloader will be affected as well.

Reply:
Fixed in 06bd726ab312008d3d54f5ac214c0180975e0ad6.